### DISCUSSION TITLE
Getting Started with GitHub Part 2: Best practices for notifications

### DISCUSSION THREAD

**Discussion by potaders** [community/community#47569] [Discover: GitHub Best Practices]
GitHub is where the world creates, stores, and collaborates on code. In this article, we'll explore how GitHub users can keep track of what they're working on and what interests them, through GitHub notification settings.

#### Types of notifications

There are many functions of GitHub that can send notifications, but only two ways to opt into them: participating and subscribing.

Participation notifications come from events that directly involve something attached to you. These notifications could come from a pull request opened or assigned to you, an issue you opened, or a discussion topic you created. These are a few examples; and a exhaustive list can be found in our [documentation](https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/setting-up-notifications/about-notifications#default-subscriptions).

Subscription notifications come from events that you choose to opt into. For example, you can subscribe to an open source repo by watching a repository, subscribing to a specific GitHub issue, or subscribing to security alerts in a repository. More examples can be found [here](https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/setting-up-notifications/about-notifications#subscription-options).

To view what kind of notifications you will see from a repository, navigate to the repo root page. At the top right of the page there is a dropdown labeled `Watch`, clicking this will show what kind of notifications you'll receive from this repo. `Participating and @mentions` are a the default setting, `All Activity` can be noisy, and `Ignore` will send you nothing. More on `Custom` notifications can be found [here](https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/setting-up-notifications/configuring-notifications#configuring-your-watch-settings-for-an-individual-repository).

![Watching-dropdown](https://user-images.githubusercontent.com/103224897/219085496-650c9307-794b-4810-91a5-2bdab6dd74d6.png)

#### GitHub Notification Delivery Options

GitHub can send notifications in three ways: through a browser in GitHub.com, through the Github Mobile app, and through email. As a user of Github, you can choose to receive these notifications in any combination of these options. Documentation on how to customize your notification settings can be found [here](https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/setting-up-notifications/configuring-notifications#choosing-your-notification-settings).

GitHub notifications have a shared state. If you open a notification sent via email, it will mark a web or mobile notification as 'read' in the browser or mobile app. In the next section we'll talk about how you can update your preferences around notifications.

Where I prefer to view my GitHub notifications is the [notification inbox](https://github.com/notifications). The link to this is in the upper right corner of any GitHub.com page, next to the search bar and your profile picture icon. The notification icon will have a dot over it if you have any notifications!
![Notification-icon](https://user-images.githubusercontent.com/103224897/219085674-b25a91b6-aae8-4234-9b31-5718268a2b2d.png)

In the inbox you can view all active and viewed notifications. Viewing, filtering, and grouping notifications can all be done on this page. Each notification will tell you what repo it's coming from, along with why you received it, and how long ago it arrived. Notifications can be cleared by selecting the checkbox next to them and marking `Done` to continue receiving updates on this topic, or `Unsubscribe` to clear the notification and no longer receive updates.

For example, say you receive a notification about a Pull Request your GitHub Team was mentioned in for feedback. You are interested in learning more, but don't need to see the notification anymore, so you clear it with `Done`. In another example, you might receive a notification for an Issue but it's not something you're working on or are interested in. You can clear the notification via `Unsubscribe` to no longer see updates on this issue.

### Notification Preferences

No matter how you use GitHub, the easiest way to update preferences is on a Github.com. To get to your notification settings, click on your profile icon in the upper right corner and then click `Settings`. On the left side of the settings page you should see a `Notifications` section, click that to view your settings.

As a first step, follow [this guide](https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/setting-up-notifications/configuring-notifications) to choose your delivery methods—-if any, for participating and watching notifications.

#### Email Settings

If you decide to enable email notifications, you'll have a few more choices to make. On the left of the page, click `Emails` to update those settings.

When creating a GitHub account, you needed to provide an email, you should see that email here along with any other email addresses you've added to GitHub. More documentation on how these settings work can be found [here](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences), now let's return to the `Notification` settings page.

The first thing you should see on this page is selecting a default email address. Take note that this email address is the same as the primary email address we saw on the `Emails` settings page. The dropdown on this portion of this setting allows you to select any other email address connected to your GitHub account to receive notifications.
![Email-notifications](https://user-images.githubusercontent.com/103224897/219085781-ddfa3e5b-50e7-42fe-b939-90ebd8288c34.png)

Additionally, you can set up custom routing of email notifications based on each GitHub organization you are a part of. This can be extremely helpful if you want to direct notifications from your work organization to your work email, and notifications of a personal organization to your personal email.

#### Watching Settings

In the next section, you can make two choices about automatic watching of repositories. As we talked about earlier, one of the easiest ways to get notifications is to 'Watch' a repository. 

- Automatically watch repositories
- Automatically watch teams

Both of these settings will automatically set you up to receive notifications. Watch repositories will opt you into notifications if you push code to that repo, and watch teams will opt you in if you are a member of a GitHub Team that is @mentioned in a repository.

#### Subscriptions

We talked earlier about the three places where notifications show up, in this dropdown you can choose to receive notifications on GitHub, email, or both. Note that you don't see the GitHub mobile app here, those notifications will be sent through the 'on GitHub' selection. 

Along with Watching, you can make a choice about how you receive Participating, @mentions, and custom subscriptions. Like watching, GitHub mobile notifications are also driven through the 'on GitHub' selection.

When it comes to how you receive notifications, it's best to have them sent to a space that's easy for you to organize. For some, email is perfect because it allows you to use email filtering to sort information. For others, seeing them in a web browser is great because they are all in one spot and don't clutter and email inbox. Play around with both and see which you prefer!

#### System Notifications

In this section you can make choices about how you receive notifications related to other functions of GitHub. More information on Dependabot alerts settings can be found [here](https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/setting-up-notifications/configuring-notifications#dependabot-alerts-notification-options). More information on GitHub Actions alert settings can be found [here](https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/setting-up-notifications/configuring-notifications#github-actions-notification-options).

Thanks for spending time to learn about how GitHub notifications work!
For questions related to GitHub notifications please refer to our documentation as a first step. If you have further questions, reach out to the community! Everyone uses GitHub in their own way and we're happy to help answer your questions.

**Update!** 
Links to the other parts of this of this series:
- [Part 1](https://github.com/orgs/community/discussions/39082)
- [Part 3](https://github.com/orgs/community/discussions/72264)

**Comment by joanna7456** [comment]
Jestem nowa na Githubie. Czy ktoś mi troszkę pomoże. Joanna

**Reply by queenofcorgis** [reply]
Welcome to GitHub and the GitHub Community @joanna7456  🎆 ! We're glad you're here and can't wait to see where your GitHub journey takes you. Check out this [check-in](https://github.com/orgs/community/discussions/42302#discussion-4683140) with links on how to choose your first project to contribute to, a blog with getting started resources, and a new user checklist.

Best of luck!

**Comment by Abdoelsabahy** [comment]
Thanks

**Comment by olaitan-blip** [comment]
Good day to you all, thank you for the opportunity to be here.  I wish to let you know i am a beginner, i know nothing about coding, i will need guide on how to start.
Secondly, i have a large dataset to use for my research work, i have been having problems on how to separate my variables into distinct columns in excel, into year 2019, 2020 and 2021, pls, find attached below,
[xix-file.xlsx](https://github.com/community/community/files/10753205/xix-file.xlsx)
 , i will appreciate from this community anyone who can help, thank you

**Reply by tuves** [reply]
Hi @olaitan-blip,

Thanks for being a part of the GitHub Community, we're glad you're here!

If you're looking for help for this specific topic, you might want to try asking for help somewhere that focuses on Microsoft Excel. It's possible that another GitHub user might have run into this same issue and can help, but the GitHub Community on Discussions focuses primarily on topics related to GitHub itself or collaboration on project development and ideas. Specifically, this blog is about GitHub notifications. We want to make sure you’re getting the best support you can, but this space may not be the right place for this particular topic.

Best of luck!

**Reply by GPI67715** [reply]
When, as a beginner, feel comfortable that your VS Code is ready to start coding independently with collaboration with GitHub Community.

**Comment by Habnix69mk** [comment]
Hallo!
Ich habe mich bei GitHub angemeldet, weil ich mit Linux bzw. Archlinux vertraut bin. Außerdem interessiert mich alles was mit Betriebssystemen zu Tun hat. Bin aber schon Neuling hier in GitHub und will jetzt nach und nach schauen, was ich hier finden kann, um mein Hobby(PC, OS, Bootmanager und verschiedene Programme) benutzen zu können. Habe zwar auch vieles was Geld kostet aber am Liebsten schaue ich nach alternativen. Opensource Freeware kostenlose Betriebssysteme. Neu und Alt. Ich habe auch noch eine Floppy 3.5" USB und externer DVD Brenner, Laptop, PC, Smartphone!

Ansonsten viele Grûße an das GitHub Team und an alle die Hier sind. 
Mit freundlichen Grüßen
(_removed by moderator_)

**Reply by tuves** [reply]
Hi @Habnix69mk, thanks for participating!
We take our [Code of Conduct](https://github.com/community/community/blob/0ca37cf594e438fd4aec3c7f3736b95e6b0b3567/CODE_OF_CONDUCT.md) very seriously and want to help ensure that everyone has a pleasant experience free of antagonism and harassment. Unfortunately, we don’t currently have moderators for languages other than English. Until that changes, we need to ask that everyone use English here in the GitHub Community when posting. Furthermore, we do not allow PII in posts, so I've amended your post.

**Comment by Habnix69mk** [comment]
OK i have understand you, thanks a Lot!

tuves ***@***.***> schrieb am Do., 16. Feb. 2023, 12:36:

**Comment by IvenLagowan** [comment]
web

**Comment by jeanp1209** [comment]
> GitHub est l'endroit où le monde crée, stocke et collabore sur le code. Dans cet article, nous allons explorer comment les utilisateurs de GitHub peuvent garder une trace de ce sur quoi ils travaillent et de ce qui les intéresse, via les paramètres de notification GitHub.
> 
> ### Comprendre les notifications
> #### Types de notification
> Il existe de nombreuses fonctions de GitHub qui peuvent envoyer des notifications, mais il n'y a que deux façons de s'y inscrire : participer et s'abonner.
> 
> Les notifications de participation proviennent d'événements qui impliquent directement quelque chose qui vous est attaché. Ces notifications peuvent provenir d'une demande d'extraction ouverte ou qui vous est attribuée, d'un problème que vous avez ouvert ou d'un sujet de discussion que vous avez créé. Ce sont quelques exemples; et une liste exhaustive se trouve dans notre [documentation](https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/setting-up-notifications/about-notifications#default-subscriptions) .
> 
> Les notifications d'abonnement proviennent d'événements auxquels vous choisissez de participer. Par exemple, vous pouvez vous abonner à un référentiel open source en surveillant un référentiel, en vous abonnant à un problème GitHub spécifique ou en vous abonnant aux alertes de sécurité dans un référentiel. Plus d'exemples peuvent être trouvés [ici](https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/setting-up-notifications/about-notifications#subscription-options) .
> 
> Pour afficher le type de notifications que vous verrez à partir d'un référentiel, accédez à la page racine du référentiel. En haut à droite de la page, il y a une liste déroulante intitulée `Watch`, cliquez dessus pour afficher le type de notifications que vous recevrez de ce dépôt. `Participating and @mentions`sont le paramètre par défaut, `All Activity`peuvent être bruyants et `Ignore`ne vous enverront rien. `Custom`Vous trouverez plus d'informations sur les notifications [ici](https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/setting-up-notifications/configuring-notifications#configuring-your-watch-settings-for-an-individual-repository) .
> 
> ![Liste déroulante de surveillance](https://user-images.githubusercontent.com/103224897/219085496-650c9307-794b-4810-91a5-2bdab6dd74d6.png)
> 
> #### Options de livraison des notifications GitHub
> GitHub peut envoyer des notifications de trois manières : via un navigateur sur GitHub.com, via l'application Github Mobile et par e-mail. En tant qu'utilisateur de Github, vous pouvez choisir de recevoir ces notifications dans n'importe quelle combinaison de ces options. La documentation sur la façon de personnaliser vos paramètres de notification peut être trouvée [ici](https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/setting-up-notifications/configuring-notifications#choosing-your-notification-settings) .
> 
> Les notifications GitHub ont un état partagé. Si vous ouvrez une notification envoyée par e-mail, une notification Web ou mobile sera marquée comme "lue" dans le navigateur ou l'application mobile. Dans la section suivante, nous verrons comment vous pouvez mettre à jour vos préférences concernant les notifications.
> 
> Là où je préfère voir mes notifications GitHub, c'est la [boîte de réception des notifications](https://github.com/notifications) . Le lien vers ceci se trouve dans le coin supérieur droit de n'importe quelle page GitHub.com, à côté de la barre de recherche et de l'icône de votre photo de profil. L'icône de notification sera surmontée d'un point si vous avez des notifications ! ![Icône de notification](https://user-images.githubusercontent.com/103224897/219085674-b25a91b6-aae8-4234-9b31-5718268a2b2d.png)
> 
> Dans la boîte de réception, vous pouvez afficher toutes les notifications actives et consultées. L'affichage, le filtrage et le regroupement des notifications peuvent tous être effectués sur cette page. Chaque notification vous indiquera de quel référentiel elle provient, ainsi que la raison pour laquelle vous l'avez reçue et depuis combien de temps elle est arrivée. Les notifications peuvent être effacées en cochant la case à côté d'elles et en marquant `Done`pour continuer à recevoir des mises à jour sur ce sujet, ou `Unsubscribe`pour effacer la notification et ne plus recevoir de mises à jour.
> 
> Par exemple, supposons que vous receviez une notification concernant une demande d'extraction dans laquelle votre équipe GitHub a été mentionnée pour obtenir des commentaires. Vous souhaitez en savoir plus, mais vous n'avez plus besoin de voir la notification, vous l'effacez donc avec `Done`. Dans un autre exemple, vous pouvez recevoir une notification pour un problème, mais ce n'est pas quelque chose sur lequel vous travaillez ou qui ne vous intéresse pas. Vous pouvez effacer la notification via `Unsubscribe`pour ne plus voir les mises à jour sur ce problème.
> 
> ### Préférences de notification
> Quelle que soit la façon dont vous utilisez GitHub, le moyen le plus simple de mettre à jour les préférences est sur un Github.com. Pour accéder à vos paramètres de notification, cliquez sur l'icône de votre profil dans le coin supérieur droit, puis cliquez sur `Settings`. Sur le côté gauche de la page des paramètres, vous devriez voir une `Notifications`section, cliquez dessus pour afficher vos paramètres.
> 
> As a first step, follow [this guide](https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/setting-up-notifications/configuring-notifications) to choose your delivery methods—-if any, for participating and watching notifications.
> 
> #### Email Settings
> If you decide to enable email notifications, you'll have a few more choices to make. On the left of the page, click `Emails` to update those settings.
> 
> When creating a GitHub account, you needed to provide an email, you should see that email here along with any other email addresses you've added to GitHub. More documentation on how these settings work can be found [here](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences), now let's return to the `Notification` settings page.
> 
> The first thing you should see on this page is selecting a default email address. Take note that this email address is the same as the primary email address we saw on the `Emails` settings page. The dropdown on this portion of this setting allows you to select any other email address connected to your GitHub account to receive notifications. ![Notifications par email](https://user-images.githubusercontent.com/103224897/219085781-ddfa3e5b-50e7-42fe-b939-90ebd8288c34.png)
> 
> Additionally, you can set up custom routing of email notifications based on each GitHub organization you are a part of. This can be extremely helpful if you want to direct notifications from your work organization to your work email, and notifications of a personal organization to your personal email.
> 
> #### Watching Settings
> In the next section, you can make two choices about automatic watching of repositories. As we talked about earlier, one of the easiest ways to get notifications is to 'Watch' a repository.
> 
> * Automatically watch repositories
> * Automatically watch teams
> 
> Both of these settings will automatically set you up to receive notifications. Watch repositories will opt you into notifications if you push code to that repo, and watch teams will opt you in if you are a member of a GitHub Team that is @mentioned in a repository.
> 
> #### Subscriptions
> We talked earlier about the three places where notifications show up, in this dropdown you can choose to receive notifications on GitHub, email, or both. Note that you don't see the GitHub mobile app here, those notifications will be sent through the 'on GitHub' selection.
> 
> Along with Watching, you can make a choice about how you receive Participating, @mentions, and custom subscriptions. Like watching, GitHub mobile notifications are also driven through the 'on GitHub' selection.
> 
> When it comes to how you receive notifications, it's best to have them sent to a space that's easy for you to organize. For some, email is perfect because it allows you to use email filtering to sort information. For others, seeing them in a web browser is great because they are all in one spot and don't clutter and email inbox. Play around with both and see which you prefer!
> 
> #### System Notifications
> In this section you can make choices about how you receive notifications related to other functions of GitHub. More information on Dependabot alerts settings can be found [here](https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/setting-up-notifications/configuring-notifications#dependabot-alerts-notification-options). More information on GitHub Actions alert settings can be found [here](https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/setting-up-notifications/configuring-notifications#github-actions-notification-options).
> 
> Thanks for spending time to learn about how GitHub notifications work! For questions related to GitHub notifications please refer to our documentation as a first step. If you have further questions, reach out to the community! Everyone uses GitHub in their own way and we're happy to help answer your questions.

**Comment by jeanp1209** [comment]
https://github.com/orgs/community/discussions/47569#discussion-4862324

**Comment by saraboosh79** [comment]
![Screenshot_20230311-064124](https://user-images.githubusercontent.com/94699114/224485705-74794a15-4d76-435a-aac5-9345b2770890.png)

**Reply by saraboosh79** [reply]
I paid all this stuff I paid a 64 bucks plus two more transaction and I never got money for it back

**Comment by jobrien12345** [comment]
I am attempting to submit a release.  The file has been loaded.  But it will not publish.
